backends/bmv2/common/action.cpp

```diff
@@ -129,7 +129,7 @@ void ActionConverter::convertActionBody(const IR::Vector<IR::StatOrDecl>* body,
                 continue;
             }
         }
-        ::error("%1%: not yet supported on this target", s);
+        ::error(ErrorType::ERR_EXPRESSION, s, "not yet supported on this target");
```

Comment:
I find these errors harder to understand and write for the developer.
Wouldn't it be easier to add an enum to the error function describing the category?

Reply:
That's what ERR_EXPRESSION is (well, not really an enum because of C++ limitations on extending enum classes, and thus forcing me to use a class with static const members). In addition to the enum, this proposal structures the error so that we have a program object that we can use as source. If you prefer the original messages (typos included for some of them! :-) ) we can keep the messages and simply use the format from here rather than the error catalog.